myia/abstract/aliasing.py

```diff
@@ -41,6 +38,16 @@ def _explore(self, v: object, vseq, path):
         vseq = (*vseq, v)
         for k, x in dataclass_fields(v).items():
             yield from self(x, vseq, (*path, k))
+    else:
+        vseq = (*vseq, v)
+
+        try:
+            fields = get_fields(v)
+        except TypeError:
+            return None
+
+        for k, x in fields.items():
+            yield from self(x, vseq, (*path, k))
```

Comment:
Since `get_fields` also works for dataclasses, you can replace the entire body of this overload by the contents of the else branch (you don't need to check for is_dataclass).